dhcpv4/ztp/ztp.go

```diff
@@ -0,0 +1,139 @@
+package ztpv4
+
+import (
+	"bytes"
+	"errors"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// VendorData is optional data a particular vendor may or may not include
+// in the Vendor Class options.  All values are optional and will be zero
+// values if not found.
+type VendorData struct {
+	VendorName string
+	Model      string
+	Serial     string
+}
+
+var errVendorOptionMalformed = errors.New("malformed vendor option")
+
+// VendorDataV4 will try to parse dhcp4 options data looking for more specific
+// vendor data (like model, serial number, etc).  If the options are missing
+func VendorDataV4(packet *dhcpv4.DHCPv4) VendorData {
+	vd := VendorData{}
+
+	if err := parseV4VendorClass(&vd, packet); err != nil {
+		glog.Errorf("failed to parse vendor data from vendor class: %v", err)
+	}
+
+	if err := parseV4VIVC(&vd, packet); err != nil {
+		glog.Errorf("failed to parse vendor data from vendor-idenitifying vendor class: %v", err)
+	}
+
+	return vd
+}
+
+// parseV4Opt60 will attempt to look at the Vendor Class option (Option 60) on
```

Comment:
This comment omits an important thing about what the function does. This function modifies *in-place* the input object. I'm not a fan of this approach because it is less obvious to understand what it does, by its name. Wondering if we should instead return a copy with the modified fields along with an error